### ISSUE TITLE
Proxy settings (requestDefaults) only handles the host, not the full proxy URL

### ISSUE THREAD

**Issue by sdickson-sv** [dscape/nano#251]
Hi,

I'm having an issue trying use nano via a proxy API. I've set up my nano object like this:
```javascript
var apiEndpoint = require('nano')(
  { "url"             : "http://localhost:5984/couchInstance"
  , "requestDefaults" : { "proxy" : "http://localhost:54321/api/document/couchDocument" }
  });
```
If I make a call to get a view back:
```javascript
apiEndpoint.view('foo', 'bar', params, function (err, body){ ... }
```
it is not going to the URL I want.

The URL I was expecting it to hit was:
http://localhost:54321/api/document/couchDocument/_design/foo/_view/bar?limit=26

But instead it is hitting this URL:
http://localhost:54321/couchInstance/_design/foo/_view/bar?limit=26

The proxy setting is only swapping out the host i.e. swapping localhost:5984 for localhost:54321. Is this just down to how request.js works?

**Comment by jo** [comment]
Yes, that seems the way [requests proxies](https://github.com/request/request#proxies) work.
You can see in [here](https://github.com/request/request/blob/master/request.js#L440) and [here](https://github.com/request/request/blob/master/request.js#L543) that in case of a http proxy only the hostname and port are swapped.

**Comment by sdickson-sv** [comment]
Thanks.

I've looked into this a bit further. What i'm finding is that when my cfg.url is being setup part of the URL is being replaced.

e.g. I set the url to point to my proxy api for CouchDb like this
```javascript
var apiEndpoint = require('nano')(
  { "url"             : "http://localhost:54321/api/document/couchDocument"  });
```
Then in nano.js around line 722 there is this code which is removing the 'couchDocument' part of my URL:
```javascript
  var rootPath = path.pathname.replace(/\/?$/, '/..');

  if (db) {
    cfg.url = u.resolve(cfg.url, rootPath).replace(/\/?$/, '');
    return docModule(db);
  }
```

looking at the last commit this line was introduced. Is this to enable accessing CouchDb from a relative path?
```javascript
  var rootPath = path.pathname.replace(/\/?$/, '/..');
```
I don't know if this would be acceptable or would cause further issues, but i've found this fixes the problem i'm seeing.
```javascript
var rootPath = path.pathname.replace(/\/?$/, '/.');
```

Your thoughts on this would be appreciated.

**Comment by jo** [comment]
On initialisation nano inspects the url and if there are path components, the last one is considered the database name. I see that in your case this behaviour is not expected.

Also this questions nanos ability to access CouchDBs from subdirectories, which was introduced in the [abovementioned commit](https://github.com/dscape/nano/commit/76004b230bfeff5862bb491f6ea23399ed1fce90).

Nano is indeed able to use a subdirectory but only if you are configuring nano directly with a database, like this:
```js
require('nano')('http://localhost:5984/path/to/couchdb/dbname')
```
Maybe thats what you need?

However - I see the need for an option which controls said behaviour, maybe a `parseUrl` flag. I will think about it. @dscape any ideas?

**Comment by sdickson-sv** [comment]
Thanks, yes that does work for me, I don't know why I didn't try that earlier. So just to round out this discussion:

This doesn't work
```javascript
var apiEndpoint = require('nano')(
  { "url"             : "http://localhost:54321/api/document/couchDocument"  });
```
This does
```javascript
var apiEndpoint = require('nano')('http://localhost:54321/api/document/couchDocument');
```
I'm able to progress now, thanks for your help on this.

**Comment by dscape** [comment]
This is great. Parseurl flag is a good idea imho.

**Comment by jo** [comment]
@sdickson-sv it does not make a difference if you specify the url under an objects `url` key or as string parameter.

```js
require('nano')('http://localhost:54321/api/document/couchDocument')
```
is equivalent to
```js
require('nano')({
  url: 'http://localhost:54321/api/document/couchDocument' 
})
```

**Comment by jo** [comment]
Pull request for `parseUrl` flag: #254

**Comment by jo** [comment]
fixed with `6.1.0`. Please refer to the [configuration section](https://github.com/dscape/nano#configuration) for details.
